<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Massimo Callegari</author>
 <comment>
  Q Light Controller Plus
  inputselectionwidget.ui

  Copyright (c) 2015 Massimo Callegari

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0.txt

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
 </comment>
 <class>InputSelectionWidget</class>
 <widget class="QWidget" name="InputSelectionWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>504</width>
    <height>185</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Form</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <property name="verticalSpacing">
    <number>0</number>
   </property>
   <item row="0" column="1">
    <widget class="QGroupBox" name="m_keyInputGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Key combination</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="leftMargin">
       <number>3</number>
      </property>
      <property name="topMargin">
       <number>4</number>
      </property>
      <property name="rightMargin">
       <number>3</number>
      </property>
      <property name="bottomMargin">
       <number>4</number>
      </property>
      <item row="1" column="2">
       <widget class="QToolButton" name="m_detachKey">
        <property name="toolTip">
         <string>Reset the keyboard shortcut key</string>
        </property>
        <property name="text">
         <string notr="true">...</string>
        </property>
        <property name="icon">
         <iconset resource="qlcui.qrc">
          <normaloff>:/fileclose.png</normaloff>:/fileclose.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QToolButton" name="m_attachKey">
        <property name="toolTip">
         <string>Set a key combination for this widget</string>
        </property>
        <property name="text">
         <string notr="true">...</string>
        </property>
        <property name="icon">
         <iconset resource="qlcui.qrc">
          <normaloff>:/key_bindings.png</normaloff>:/key_bindings.png</iconset>
        </property>
        <property name="iconSize">
         <size>
          <width>32</width>
          <height>32</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="2">
       <widget class="QLineEdit" name="m_keyEdit">
        <property name="toolTip">
         <string>Keyboard combination that toggles this widget</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QGroupBox" name="m_extInputGroup">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>External Input</string>
     </property>
     <layout class="QGridLayout" name="gridLayoutExternalInput">
      <property name="leftMargin">
       <number>5</number>
      </property>
      <property name="topMargin">
       <number>4</number>
      </property>
      <property name="rightMargin">
       <number>5</number>
      </property>
      <property name="bottomMargin">
       <number>4</number>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Input Universe</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" colspan="4">
       <widget class="QLineEdit" name="m_inputUniverseEdit">
        <property name="toolTip">
         <string>The input universe that sends data to this widget</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="5">
       <widget class="QGroupBox" name="m_feedbackGroup">
        <property name="title">
         <string>Custom feedback</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <property name="topMargin">
          <number>3</number>
         </property>
         <property name="bottomMargin">
          <number>3</number>
         </property>
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Lower value</string>
           </property>
          </widget>
         </item>
         <item row="0" column="3">
          <widget class="QSpinBox" name="m_upperSpin">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>255</number>
           </property>
           <property name="value">
            <number>255</number>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="m_lowerSpin">
           <property name="maximum">
            <number>255</number>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Upper value</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="1" column="1" colspan="4">
       <widget class="QLineEdit" name="m_inputChannelEdit">
        <property name="toolTip">
         <string>The particular input channel within the input universe that sends data to this widget</string>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Input Channel</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0" colspan="5">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QToolButton" name="m_customFbButton">
          <property name="text">
           <string>Custom Feedback</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="m_autoDetectInputButton">
          <property name="toolTip">
           <string>When toggled, you can click an external button to assign it to this widget.</string>
          </property>
          <property name="text">
           <string>Auto Detect</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>false</bool>
          </property>
          <property name="flat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="m_chooseInputButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Choose an external input universe &amp; channel that this widget should listen to.</string>
          </property>
          <property name="text">
           <string>Choose...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="qlcui.qrc"/>
 </resources>
 <connections/>
</ui>
